Common/tests/Submitter/IntegrationGrpcSubmitterServiceTest.cs

```diff
@@ -148,109 +146,198 @@ public async Task TryGetResultShouldSucceed()
-                        submitter => submitter.CreateSession(It.IsAny<IList<string>>(),
-                                                             It.IsAny<TaskOptions>(),
-                                                             It.IsAny<CancellationToken>()),
-                        submitter => submitter.CreateTasks(It.IsAny<string>(),
-                                                           It.IsAny<string>(),
-                                                           It.IsAny<TaskOptions>(),
-                                                           It.IsAny<IAsyncEnumerable<gRPC.Services.TaskRequest>>(),
-                                                           It.IsAny<CancellationToken>()),
-
-                        submitter => submitter.WaitForCompletion(It.IsAny<WaitRequest>(),
-                                                                 It.IsAny<CancellationToken>()),
-                        submitter => submitter.TryGetTaskOutputAsync(It.IsAny<TaskOutputRequest>(),
-                                                                     It.IsAny<CancellationToken>()),
-                        submitter => submitter.WaitForAvailabilityAsync(It.IsAny<ResultRequest>(),
-                                                                        It.IsAny<CancellationToken>()),
-                        submitter => submitter.GetTaskStatusAsync(It.IsAny<GetTaskStatusRequest>(),
-                                                                  It.IsAny<CancellationToken>()),
-                        submitter => submitter.GetResultStatusAsync(It.IsAny<GetResultStatusRequest>(),
-                                                                    It.IsAny<CancellationToken>()),
-                        submitter => submitter.ListTasksAsync(It.IsAny<TaskFilter>(),
-                                                              It.IsAny<CancellationToken>()),
-                      };
-
-    foreach (var expression in expressions)
-    {
-      mockSubmitter.Setup(expression)
-                   .Throws(exception);
-    }
-
-    return mockSubmitter.Object;
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+  // we need this async because only with async methods, the exceptions are not properly catched by gRPC
+  // and the error codes are not the ones we specified
+  public class AsyncThrowExceptionSubmitter<T> : ISubmitter
+    where T : Exception, new()
+  {
+    public async Task CancelSession(string            sessionId,
+                                    CancellationToken cancellationToken)
+      => throw new T();
+
+    public async Task CancelTasks(TaskFilter        request,
+                                  CancellationToken cancellationToken)
+      => throw new T();
+
+    public async Task<Count> CountTasks(TaskFilter        request,
+                                        CancellationToken cancellationToken)
+      => throw new T();
```

Comment:
You can remove the async, and return a Task created from an exception:

`Task.FromException<Count>(new T())`